utilities/workers/worker_manager.py

```diff
@@ -2,38 +2,183 @@
+from utilities.workers import queue_proxy_wrapper
+
+
+class WorkerProperties:
+    """
+    Worker Properties.
+    """
+
+    __create_key = object()
+
+    @classmethod
+    def create(
+        cls,
+        count: int,
+        target: "(...) -> object",  # type: ignore
+        work_arguments: "tuple",
+        input_queues: "list[queue_proxy_wrapper.QueueProxyWrapper]",
+        output_queues: "list[queue_proxy_wrapper.QueueProxyWrapper]",
+        controller: worker_controller.WorkerController,
+        local_logger: logger.Logger,
+    ) -> "tuple[bool, WorkerProperties | None]":
+        """
+        Creates worker properties.
+
+        count: Number of workers.
+        target: Function.
+        work_arguments: Arguments for worker internals.
+        input_queues: Input queues.
+        output_queues: Output queues.
+        controller: Worker controller.
+        local_logger: Existing logger from process.
+
+        Returns the WorkerProperties object.
+        """
+        if count <= 0:
+            frame = inspect.currentframe()
+            local_logger.error(
+                "Worker count requested is less than or equal to zero, no workers were created",
+                frame,
+            )
+            return False, None
+
+        return True, WorkerProperties(
+            cls.__create_key,
+            count,
+            target,
+            work_arguments,
+            input_queues,
+            output_queues,
+            controller,
+        )
+
+    def __init__(
+        self,
+        class_private_create_key: object,
+        count: int,
+        target: "(...) -> object",  # type: ignore
+        work_arguments: "tuple",
+        input_queues: "list[queue_proxy_wrapper.QueueProxyWrapper]",
+        output_queues: "list[queue_proxy_wrapper.QueueProxyWrapper]",
+        controller: worker_controller.WorkerController,
+    ) -> None:
+        """
+        Private constructor, use create() method.
+        """
+        assert class_private_create_key is WorkerProperties.__create_key, "Use create() method"
+
+        self.__count = count
+        self.__target = target
+        self.__work_arguments = work_arguments
+        self.__input_queues = input_queues
+        self.__output_queues = output_queues
+        self.__controller = controller
+
+    def get_worker_arguments(self) -> "tuple":
+        """
+        Concatenates the worker properties into a tuple.
+
+        Returns the worker properties as a tuple.
+        """
+        return (
+            self.__work_arguments
+            + tuple(self.__input_queues)
+            + tuple(self.__output_queues)
+            + (self.__controller,)
+        )
+
+    def get_worker_count(self) -> int:
+        """
+        Returns the worker count.
+        """
+        return self.__count
+
+    def get_worker_target(self) -> "(...) -> object":  # type: ignore
+        """
+        Returns the worker target.
+        """
+        return self.__target
```

Comment:
let's follow the design pattern of the other structs, instead of having getter functions, directly reference the variables

Reply:
I disagree, this is a stateful machine, rather than a data exchange struct.